src/main.rs

```diff
@@ -70,7 +67,15 @@ impl VM {
             let instruction = (instruction >> 12) & 0xF;
             let opcode: Opcode = Opcode::from(instruction);
 
-            self.execute(opcode, instruction)?;
+            match self.execute(opcode, instruction) {
+                // Continue running if Ok or if error is not Halt
+                Ok(_) => continue,
+                Err(VMError::TrapError(TrapError::Halt)) => {
+                    println!("Program terminated normally.");
+                    return Ok(());
+                }
```

Comment:
Why is Halt an error? This works but looks wrong.

Reply:
I agree with Lean here, Halting is not an error per se. 

Maybe it's just semantics, but having a program "Halt" is part of the "intended" behavior I feel like

Reply:
Thanks Lean i understand, i used the Error enum and the handled it in the main function to gracefully halt the program, is this wrong semantically?